cmd/federation-controller/main.go

```diff
@@ -212,6 +212,7 @@ func main() {
 			go func() {
 				if err := federationClient.Run(); err != nil {
 					log.Fatalf("failed to start FDS client: %v", err)
+					time.AfterFunc(5*time.Second, federationClient.Restart)
```

Comment:
I wonder if it would be worth adding a configurable retry count to `Restart` so that we can exit with an error instead of failing indefinitely?
```suggestion
					// Configurable retry count added to signature of Restart
					if err := federationClient.Restart(5); err != nil {
					    // Log error and exit with non-zero status 
					}
```

Reply:
That makes sense. Retrying indefinitely seems like a better option. Related thought: Maybe in the future we could add a status condition to the CRD indicating the connection status so that users would not need to check logs to see that the connection has failed.